templates/manila/bin/healthcheck.py

```diff
@@ -0,0 +1,227 @@
+#!/usr/bin/env python3
+#
+# Copyright 2023 Red Hat Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+# Trivial HTTP server to check health of scheduler, and share services.
+# Manila-API has its own health check endpoint and does not need this.
+#
+# The only check this server currently does is using the manila-manage
+# service list command, which accesses DB directly and gets data related
+# to the services already registered in Manila.
+#
+# Manila-manage will allow us to keep imports minimal and also check on
+# what has already been registered in the manila-database, and with this
+# we can checks if services are up or not.
+#
+# For share services all enabled backends must be up to return 200, so it is
+# recommended to use a different pod for each backend to avoid one backend
+# affecting others.
+#
+# Requires the name of the service as the first argument (share,
+# scheduler) and optionally a second argument with the location of the
+# configuration directory (defaults to /etc/manila/manila.conf.d)
+
+from http import server
+import logging
+import signal
+import subprocess
+import sys
+import time
+import threading
+
+from oslo_config import cfg
+
+from manila.share import configuration as share_conf
+
+
+HOSTNAME = ''
+SERVER_PORT = 8080
+CONF = cfg.CONF
+BINARIES = ('share', 'scheduler')
+STATE_UP = ":-)"
+STATUS_ENABLED = "enabled"
+
+
+class ServiceNotFoundException(Exception):
+    message = "Service not found."
+
+    def __init__(self, filters={}):
+        if filters:
+            self.message = f'Service not found with filters: {filters}'
+        super(ServiceNotFoundException, self).__init__(self.message)
+
+
+class ServiceNotUp(Exception):
+    message = "Service error: Service is not UP."
+
+    def __init__(self):
+        super(ServiceNotUp, self).__init__(self.message)
+
+
+class HeartbeatServer(server.BaseHTTPRequestHandler):
+    @classmethod
+    def initialize_class(cls, binary, config_dir):
+        """Calculate and initialize constants"""
+        cls.binary = 'manila-' + binary
+        cls.config_dir = config_dir
+
+        if binary != 'share':
+            services_filters = [
+                {'host': CONF.host,
+                 'state': STATE_UP,
+                 'status': STATUS_ENABLED,
+                 'binary': cls.binary}]
+        else:
+            backend_opts = [
+                cfg.StrOpt('backend_host'),
+                cfg.StrOpt('backend_availability_zone', default=None),
```

Comment:
what are these used for?

Reply:
Yeah, not being used anymore - I remember it was in the past, but I forgot to remove it. Thanks